src/AzureClient/AzureEnvironment.cs

```diff
@@ -58,17 +58,39 @@ public static AzureEnvironment Create(string subscriptionId)
             return Production(subscriptionId);
         }
 
+        private string GetNormalizedLocation(string location)
+        {
+            if (Type == AzureEnvironmentType.Canary)
+            {
+                return "eastus2euap";
+            }
```

Comment:
Instead of adding a hard coded condition here to check for for Canary, should we instead by passing a different BaseUriForLocation function when constructing Canary below (L224)?

Reply:
Yes, I think that's a better way to do this.